### PR TITLE
VPN-6290 - Stop using the glean sdk generator script

### PR DESCRIPTION
**Pull request by brizental**
~We should do the same for Android. Let's do it in a separate PR because this is currently breaking CI IIUC.~

Doing Android here since fixing things for iOS broke them for Android.

### CODE REVIEW

**Comment by brizental** [comment]
@strseb I am assuming now we don't need to run gradle twice in cmake.sh? From local testing it _seems_ to work -- the apk was created fine with just `cmake build --build -j${nproc}`... Let me know though.

File: android/daemon/build.gradle [hunk 1]
```diff
@@ -52,6 +41,8 @@ android {
         minSdkVersion Config.minSdkVersion
         targetSdkVersion Config.targetSdkVersion
 
+        buildConfigField "int", "VERSION_CODE", System.getenv("VERSIONCODE") ?: "99999"
+        buildConfigField "String", "VERSION_NAME", '"' + (System.getenv("SHORTVERSION") ?: "") + '"'
         buildConfigField  "String", "VERSIONCODE" ,  '"' +System.getenv("VERSIONCODE") + '"'
```
Comment (strseb) [inline android/daemon/build.gradle:46]: Why twice?

File: android/vpnClient/build.gradle [hunk 2]
```diff
@@ -67,6 +61,14 @@ android {
             includeAndroidResources = true
         }
     }
+
+    sourceSets {
+        main {
+            kotlin.srcDirs = [
+                "${System.getProperty('user.dir')}/../../clienttelemetry"
```
Comment (strseb) [inline android/vpnClient/build.gradle:68]: Not a fan of this, can you either infer that from `$rootDir` or get that copied into the project?

File: android/buildSrc/src/main/kotlin/Config.kt [hunk 3]
```diff
@@ -3,9 +3,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 object Config {
-    const val compileSdkVersion = 33
+    const val compileSdkVersion = 34
     const val buildToolsVersion = "30.0.3"
     const val minSdkVersion = 26
-    const val targetSdkVersion = 33
+    const val targetSdkVersion = 34
```
Comment (strseb) [inline android/buildSrc/src/main/kotlin/Config.kt:9]: There is one more thing to to 
https://developer.android.com/about/versions/14/behavior-changes-14#fgs-types. 

In the daemon android manifest you need to set a type: https://developer.android.com/about/versions/14/changes/fgs-types-required

File: android/daemon/build.gradle (hunk 1 above)
Comment (brizental) [reply android/daemon/build.gradle:46]: Oh, one is "VERSIONCODE" and the other is "VERSION_CODE" 👀 

It's good you commented here. It reminds me I had a little issue with these variables. 

You and the internet had told me I should use `versionName` and `versionCode` here and they would set these variables. That didn't work though, that is why I used `buildConfigField` here instead.

Also, I set these in a bunch of places, but I could not figure out how to just set it once. I don't like the way it is right now :(

File: android/vpnClient/build.gradle (hunk 2 above)
Comment (brizental) [reply android/vpnClient/build.gradle:68]: I can infer from `$rootDir`. But that would require me to assume the name of the build folder, which doesn't seem right. I also didn't want to generate these files outside of the build folder... I can copy them into the projec thought. Hadn't thought of that.

Comment (brizental) [reply android/vpnClient/build.gradle:68]: > I can infer from $rootDir. But that would require me to assume the name of the build folder, which doesn't seem right.

Neverming me being wrong here... 🤦‍♀️

File: android/settings.gradle [hunk 4]
```diff
@@ -9,11 +9,22 @@ buildCache {
...
     }
 }
+
+def repositoryRoot = "$rootDir/../../.."
+def gleanVendoredPath = "$repositoryRoot/3rdparty/glean"
+
+dependencyResolutionManagement {
+    versionCatalogs {
+        create("gleanLibs") {
+            from(files("$gleanVendoredPath/gradle/libs.versions.toml"))
```
Comment (strseb) [inline android/settings.gradle:19]: Neeeat

**Review by strseb — APPROVED** [review]

**Pull request merged**